### ISSUE TITLE
Request: Allow the use of custom data when launching a VM and attaching an OS disk

### ISSUE THREAD

**Issue by vincentdo** [Azure/azure-sdk-for-java#1741]
It seems strange that custom data is only supported when we're launching the OS disk as part of the VM creation request chain and not if we create the OS disk before hand. It seems like under the hood, Azure is launching the OS disk and then attaching it to the VM after anyway?

**Comment by anuchandy** [comment]
There is no Azure API support to set custom data to a managed disk as part of disk creation. Today it is allowed to set only through VM payload.

**Comment by vincentdo** [comment]
Right. But if you set the custom data on the VM creation request and specify an existing OS disk to be attached, the custom data is ignored which doesn't seem right.
